tests/integration.go

```diff
@@ -71,7 +78,114 @@ func dockerCmd(sb integration.Sandbox, opts ...cmdOpt) *exec.Cmd {
 	return cmd
 }
 
+func isMobyWorker(sb integration.Sandbox) bool {
+	name, hasFeature := driverName(sb.Name())
+	return name == "docker" && !hasFeature
+}
+
+func isMobyContainerdSnapWorker(sb integration.Sandbox) bool {
+	name, hasFeature := driverName(sb.Name())
+	return name == "docker" && hasFeature
+}
+
 func isDockerWorker(sb integration.Sandbox) bool {
-	sbDriver, _, _ := strings.Cut(sb.Name(), "+")
-	return sbDriver == "docker"
+	name, _ := driverName(sb.Name())
+	return name == "docker"
+}
+
+func isDockerContainerWorker(sb integration.Sandbox) bool {
+	name, _ := driverName(sb.Name())
+	return name == "docker-container"
+}
+
+func driverName(sbName string) (string, bool) {
+	name := sbName
+	var hasFeature bool
+	if b, _, ok := strings.Cut(name, "+"); ok {
+		name = b
+		hasFeature = true
+	}
+	return name, hasFeature
+}
+
+func buildkitTag() string {
+	if v := os.Getenv("TEST_BUILDKIT_TAG"); v != "" {
+		return v
+	}
+	return defaultBuildKitTag
+}
+
+var (
+	bkvers   map[string]string
+	bkversMu sync.Mutex
+)
+
+func buildkitVersion(t *testing.T, sb integration.Sandbox) string {
+	bkversMu.Lock()
+	defer bkversMu.Unlock()
+
+	if bkvers == nil {
+		bkvers = make(map[string]string)
+	}
+
+	ver, ok := bkvers[sb.Name()]
+	if !ok {
+		if isDockerWorker(sb) || isDockerContainerWorker(sb) {
+			// version only available for docker and docker-container drivers
+			out, err := inspectCmd(sb, withArgs(sb.Address()))
+			require.NoError(t, err, out)
+			for _, line := range strings.Split(out, "\n") {
+				if v, ok := strings.CutPrefix(line, "BuildKit version:"); ok {
+					ver = strings.TrimSpace(v)
+					bkvers[sb.Name()] = ver
+				}
+			}
+			if ver == "" {
+				require.Fail(t, "BuildKit version not found in inspect output: "+out)
+			}
+		} else {
+			undockBin, err := exec.LookPath("undock")
+			require.NoError(t, err, "undock not found")
+
+			destDir := t.TempDir()
+			t.Cleanup(func() {
+				os.RemoveAll(destDir)
+			})
+
+			cmd := exec.Command(undockBin, "--cachedir", "/root/.cache/undock", "--include", "/usr/bin/buildkitd", "--rm-dist", buildkitImage, destDir)
+			require.NoErrorf(t, cmd.Run(), "failed to extract buildkitd binary from %q", buildkitImage)
+
+			cmd = exec.Command(filepath.Join(destDir, "usr", "bin", "buildkitd"), "--version")
+			out, err := cmd.CombinedOutput()
+			require.NoErrorf(t, err, "failed to get BuildKit version from %q: %s", buildkitImage, string(out))
+
+			v := strings.Fields(strings.TrimSpace(string(out)))
+			if len(v) != 4 {
+				require.Fail(t, "unexpected version format: "+strings.TrimSpace(string(out)))
+			}
+			ver = v[2]
+			bkvers[sb.Name()] = ver
```

Comment:
Will use builder inspect output to get BuildKit version (specially needed for docker driver). For other drivers that don't output BuildKit version in inspect (e.g., remote), we are invoking buildkitd binary with `--version`.